framework/include/base/MooseFunctor.h

```diff
@@ -894,6 +959,13 @@ FunctorBase<T>::dot(const ElemSideQpArg & elem_side_qp, const unsigned int state
   return evaluateDot(elem_side_qp, state);
 }
 
+template <typename T>
+typename FunctorBase<T>::DotType
+FunctorBase<T>::dot(const ElemPointArg & elem_point, const unsigned int state) const
```

Comment:
These evaluate functions appear to be candidates for templates? (first argument)

Reply:
Hmmmmm ... this is an excellent point. We could make the `operator()`, `gradient`, and `dot` methods  templated methods. We cannot make the `evaluate` versions templated, however, since templated functions cannot be virtual

Reply:
Oh wait, we do have different implementations for each type of argument

Reply:
So scratch that, I don't think there too much chance for more templating. I do template `queryQpCache` on the space and temporal arguments though